dids/src/main/kotlin/ion/model/Models.kt

```diff
@@ -0,0 +1,205 @@
+package web5.dids.ion.model
+
+import com.nimbusds.jose.jwk.JWK
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonClassDiscriminator
+
+/**
+ * Represents an ION document containing public keys and services.
+ *
+ * @property publicKeys List of public keys.
+ * @property services List of services.
+ */
+@Serializable
+public data class Document(
+  val publicKeys: List<PublicKey> = emptyList(),
+  val services: List<Service> = emptyList()
+)
+
+/**
+ * Represents an ION service.
+ *
+ * @property id The service ID.
+ * @property type The service type.
+ * @property serviceEndpoint The service endpoint.
+ */
+@Serializable
+public data class Service(
+  public val id: String,
+  public val type: String,
+  public val serviceEndpoint: String
+)
+
+/**
+ * Represents a public key in the ION document.
+ *
+ * @property id The public key ID.
+ * @property type The public key type.
+ * @property publicKeyJWK The public key JWK (JSON Web Key).
+ * @property purposes List of public key purposes.
+ */
+@Serializable
+public data class PublicKey(
+  public val id: String,
+  public val type: String,
+  public val controller: String? = null,
+  @SerialName("publicKeyJwk") public val publicKeyJWK: JsonWebKey? = null,
+  public val purposes: List<PublicKeyPurpose> = emptyList()
+)
+
+/**
+ * Represents a JSON Web Key (JWK) for public keys.
+ *
+ * @property kty Key Type.
+ * @property use Key Use.
+ * @property keyOps List of Key Operations.
+ * @property alg Algorithm.
+ * @property kid Key ID.
+ * @property x5u X.509 URL.
+ * @property x5c List of X.509 Certificate Chain.
+ * @property x5t X.509 Certificate SHA-1 Thumbprint.
+ * @property x5tS256 X.509 Certificate SHA-256 Thumbprint.
+ * @property x5uHeaderParam Custom X.509 URL Header Parameter.
+ * @property x5cHeaderParam List of Custom X.509 Certificate Chain Header Parameters.
+ * @property crv Curve.
+ * @property x X Coordinate.
+ * @property y Y Coordinate.
+ * @property d D.
+ */
+@Serializable
+public data class JsonWebKey(
```

Comment:
There is no nimbus class for this?

Reply:
There is, BUT, I could get a class like the example below to serializae using `kotlinx.serialization`

```kt
@Serializable
public data class Foo(
  public val jwk : JWK
)
```


It failed when calling `Json.decodeFromString(fooStringJson)`

Reply:
Let's stick with using nimbusds `JWK`to remain consistent with the rest of the codebase